### PR TITLE
httpbakery: do not allow req.Body to be set in DoWithBody

### PR DESCRIPTION
**Pull request by rogpeppe**
I got a very confusing test failure when we were accidentally doing this.

### CODE REVIEW

File: httpbakery/client.go [hunk 1]
```diff
@@ -222,6 +222,9 @@ func (c *Client) DoWithBody(req *http.Request, body io.ReadSeeker) (*http.Respon
 }
 
 func (c *Client) doWithBody(req *http.Request, body io.ReadSeeker) (*http.Response, error) {
+	if req.Body != nil {
+		return nil, errgo.New("body unexpectedly supplied in Request")
```
Comment (mattyw) [inline httpbakery/client.go]: more explicit. req.body unexpectedly supplied in request

(also note lower case r in request)

**Comment by mattyw** [comment]
LGTM after fixes

**Comment by frankban** [comment]
:+1:

**Comment by mattyw** [comment]
LGTM

**Pull request merged**